pkg/pillar/pubsub/pubsub.go

```diff
@@ -98,9 +104,19 @@ func publishImpl(agentName string, agentScope string,
 	// We always write to the directory as a checkpoint, and only
 	// write to it when persistent is set?
 	if pub.persistent {
-		pub.dirName = PersistentDirName(name)
+		if agentName == "" {
```

Comment:
Should we support this mode at all? Shouldn't this be solved by a more generic scheme?

1) There will be only one publisher and all others are Subscribers?
2) Make the publisher start early if critical - Else, let the agent subscribe and if it has to get going, use default values??

Reply:
The problem at and is that PublishToDir is a special case, and would need special handling to enable the factory for unit testing. The purpose of this PR is to address that problem by folding PublishToDir into the mainstream Publish.
The purpose of this PR is nothing more than facilitate the factory for unit testing.